cpp/include/raft/distance/detail/kernels/gram_matrix.cuh

```diff
@@ -212,7 +212,7 @@ class GramMatrixBase {
                         int ld_out)
   {
     raft::distance::distance<raft::distance::DistanceType::L2Unexpanded, math_t, math_t, math_t>(
-      x1, x2, out, n1, n2, n_cols, stream, is_row_major);
+      raft::device_resources(stream), x1, x2, out, n1, n2, n_cols, is_row_major);
```

Comment:
I think from the public API standpoint, we want to strive to accept `device_resources` everywhere (and eventually just `raft::resources`) but things inside of `detail` are okay to accept just the individual resources they need- so long as the public API is pulling resources like streams from the `raft::resources` speified from the user and not creating their own.

Reply:
For example, I'm okay with this function accepting the stream directly since it's in the `detail` namespace. The `kernel gramm` API itself is in kind of weird state right now. We moved it over from cuml mostly because it's 1) primitives built directly from the pairwise distances API, and 2) it required just enough customization that it was hard to use the existing specializtions and thus cuml's compile time was increased. 

At some point, we do need to refactor the kernel gramm API into a set of flattened stateless public API functions but for now it's kind of been left in this state just so cuml doesn't break.

Reply:
yeah - I didn't want to break this here by changing the API (for the other changes, cuml is using the pairwise_distances api's which already take a raft handle instead of a stream afaict).

This is a little weird in that its upgrading a stream -> device_resources, but since the L2Unexpanded distance here only uses the stream, this won't do anything funky like allocate a cublas handle.